x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/Range.java

```diff
@@ -149,6 +151,19 @@ public Object fold(FoldContext ctx) {
      * If they are, the value does not have to be evaluated.
      */
     protected boolean areBoundariesInvalid(Object lowerValue, Object upperValue) {
+        /*
+        NB: I am reasonably sure this code is dead.  It can only be reached from foldable(), and as far as I can tell
```

Comment:
++, I think this is a leftover from ql as we shouldn't represent datetimes as strings inside esql, not even in literals.

Reply:
That issue has been open for a year and a half.  As far as I know, it's not planned currently, and it isn't clear that we even want to add that syntax.  In the meantime, having to work around dead code adds cognitive load and extra work.  If we decide we need that syntax, and that the original logic here is important, we can always find it in the git history.

Reply:
> having to work around dead code adds cognitive load and extra work

I agree. I was thinking that knowing to look for removed code (and where) is harder than deciding we don't want this syntax and just remove the code already.

Reply:
I also think removing this dead code is the right thing to do.  I'd be happy to do so, either in this PR or in a follow up, if we're in agreement that we should.